packages/blade/src/components/Input/OTPInput/__tests__/OTPInput.web.test.tsx

```diff
@@ -95,6 +95,48 @@ describe('<OTPInput />', () => {
     expect(onOTPFilled).toHaveBeenLastCalledWith({ name: 'otp', value: otp });
   });
 
+  it('should handle onFocus', async () => {
+    const label = 'Enter OTP';
+    const onFocus = jest.fn();
+    const user = userEvent.setup();
+    const otp = '123456';
+
+    const { getAllByLabelText } = renderWithTheme(
+      <OTPInput label={label} name="otp" onFocus={onFocus} />,
+    );
+    const allInputs = getAllByLabelText(/character/);
+    await user.type(allInputs[0], otp);
+
+    // should be called for each keystroke
+    expect(onFocus).toHaveBeenCalledTimes(6);
+    expect(onFocus).toHaveBeenLastCalledWith({
+      name: 'otp',
+      value: '',
+      inputIndex: otp.length - 1,
+    });
+  });
+
+  it('should handle onBlur', async () => {
+    const label = 'Enter OTP';
+    const onBlur = jest.fn();
+    const user = userEvent.setup();
+    const otp = '123456';
+
+    const { getAllByLabelText } = renderWithTheme(
+      <OTPInput label={label} name="otp" onBlur={onBlur} />,
+    );
+    const allInputs = getAllByLabelText(/character/);
+    await user.type(allInputs[0], otp);
+
+    // should be called for each keystroke
+    expect(onBlur).toHaveBeenCalledTimes(5); // 5 because the last input will remain focused
```

Comment:
Is this expected behaviour that the user would want? 🤔 Shoudn't we treat the OTPInput as a "single input"

Since the OTPInput is considered as a "composite" of 6 smaller inputs, should the user care about all the blur events of every 6 element or should this just be called once?

Reply:
Once when? As a dev if you see 6 input fields, regardless of whether they're logically together or not, would you not expect them to have their own events?

Reply:
For instance, onChange is called for every field update, why would this logic change for onBlur and onFocus events?

Although unlikely, a user could enter the otp starting midway. Won't we still want to trigger the analytics call? 😅

Reply:
I was thinking can we put focus handlers on outer View? so if any of the fields are focussed, then we say the OTPInput is focussed. If someone clicks out, then its blur 🤔

Reply:
React Native might not behave as expected for that. View doesn't have `onBlur` or `onFocus` events, only TextInput does. Ref: https://reactnative.dev/docs/view

Reply:
I see, okay then keep it as is. Hopefully woudn’t cause any issues.